### PR TITLE
Allow spaces in image file path

### LINKED ISSUE #64: Can't set background to image with spaces in name
Version info (using the Arch Linux package):
```
$ swaylock --version
swaylock version  (Mar 11 2019, branch '')
$ pacman -Qo $(which swaylock)
/usr/bin/swaylock is owned by swaylock 1.3-3
```
Regardless of how I quote the filename in the shell, swaylock fails to open it if there are spaces present.

```
$ ls -l Running\ Man\ Nebula.jpg
-rw-r--r-- 1 josh josh 1872438 Mar  6 16:33 'Running Man Nebula.jpg'
$ swaylock -i Running\ Man\ Nebula.jpg 
2019-03-25 11:16:53 - [swaylock-1.3/background-image.c:31] Failed to load background image (Failed to open file “Running”: No such file or directory).
$ swaylock -i 'Running Man Nebula.jpg'
2019-03-25 11:19:04 - [swaylock-1.3/background-image.c:31] Failed to load background image (Failed to open file “Running”: No such file or directory).
```

### PR DESCRIPTION
**Pull request by RedSoxFan**
Fixes #64 

This allows one or more spaces to be in the image file path.

This mostly mirrors the code in sway's background subcommand

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -409,11 +426,17 @@ static void load_image(char *arg, struct swaylock_state *state) {
...
+		memmove(ptr + 1, ptr, strlen(ptr) + 1);
+		*ptr = '\\';
+	}
 	if (wordexp(image->path, &p, 0) == 0) {
 		free(image->path);
-		image->path = strdup(p.we_wordv[0]);
-		wordfree(&p);
+		image->path = join_args(p.we_wordv, p.we_wordc);
 	}
+	wordfree(&p);
```
Comment (emersion) [inline main.c]: If `wordexp` failed, `p` is undefined

**Comment by emersion** [comment]
I'm really not a fan of this, but I guess we don't really have a better solution. We could only do it when the output is specified though.

File: main.c (hunk 1 above)
Comment (RedSoxFan) [reply main.c]: Fixed

**Comment by RedSoxFan** [comment]
> We could only do it when the output is specified though.

I'm not sure I follow. Regardless if there is an output provided or not, spaces were being stripped

**Comment by emersion** [comment]
I mean, the only reason we do `wordexp` is to expand `~` when the output is specified. We could skip `wordexp` if not output prefix is specified.

**Comment by RedSoxFan** [comment]
> I mean, the only reason we do `wordexp` is to expand `~` when the output is specified. We could skip `wordexp` if not output prefix is specified.

Doesn't wordexp also allow for stuff like:
```
swaylock --image $HOME/pictures/`hostname`-lockscreen.png
swaylock --image `/path/to/random-lockscreen-image-script`

# or in the config file
image=$HOME/pictures/`hostname`-lockscreen.png
image=`/path/to/random-lockscreen-image-script`
```

**Comment by emersion** [comment]
sway's `exec` will run `sh`, which will take care of this already.

**Comment by RedSoxFan** [comment]
> sway's `exec` will run `sh`, which already takes care of this.

Not if you use the config file

**Comment by emersion** [comment]
Gah, you're right.

(Still not a fan of double-wordexp-ing things because escaping becomes hairy -- but fixing this would become complicated and probably wouldn't be worth it)

**Comment by RedSoxFan** [comment]
There could be some syntax option to use the raw path for the paths that have issues, like `[<output>:][raw:]<path>`

File: main.c [hunk 2]
```diff
@@ -409,9 +426,15 @@ static void load_image(char *arg, struct swaylock_state *state) {
 
 	// Bash doesn't replace the ~ with $HOME if the output name is supplied
 	wordexp_t p;
+	while (strstr(image->path, "  ")) {
```
Comment (emersion) [inline main.c]: Hmm, sorry if I missed something, but why are we searching for double-spaces?

File: main.c [hunk 3]
```diff
@@ -409,9 +426,15 @@ static void load_image(char *arg, struct swaylock_state *state) {
 
 	// Bash doesn't replace the ~ with $HOME if the output name is supplied
```
Comment (emersion) [inline main.c]: Could we update this comment to match the reality? (ie. we also need to do a wordexp for the config file)

While you're at it, could replace "bash" with "sh"

File: main.c (hunk 2 above)
Comment (RedSoxFan) [reply main.c]: wordexp will condense multiple spaces so stuff like `swaylock --image "path  with  multiple  spaces  in  a   row"` _(and of course, it looks like GitHub is condensing them too)_ need to have the spaces escaped.

Comment (emersion) [reply main.c]: Ehh, right

File: main.c (hunk 3 above)
Comment (RedSoxFan) [reply main.c]: Done

Comment (emersion) [reply main.c]: Have you pushed the changes?

Comment (RedSoxFan) [reply main.c]: Oh oops sorry. Apparently, I committed, but never push. Should be pushed now

**Comment by emersion** [comment]
Thanks!

**Pull request merged**